### PR TITLE
Implement KeccakWriteArgs hint

### LINKED ISSUE #285: KeccakWriteArgs
Writes arguments for the KECCAK function, splitting 64-bit integers into low and high parts.

Reference: https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/keccak_utils/keccak_utils.cairo#L26

### PR DESCRIPTION
**Pull request by TAdev0**
Resolves #285

### CODE REVIEW

**Comment by TAdev0** [comment]
@rodrigo-pino  this is a first draft. I havent implemented tests for now.

File: pkg/hintrunner/zero/zerohint_keccak.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+			if err != nil {
+				return err
+			}
+
+			var lowBig big.Int
+			var highBig big.Int
+			low.BigInt(&lowBig)
+			high.BigInt(&highBig)
+
+			var maxUint64 uint64 = math.MaxUint64
```
[the remark this answers is not in the document]
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_keccak.go:41]: oh yes, it makes sense indeed. thank you!

**Comment by TAdev0** [comment]
@rodrigo-pino  ok great, i will implement tests tonight.
Is there any other hint i can work until the next daily?

**Comment by TAdev0** [comment]
@rodrigo-pino  @cicr99  I created a draft for test, but it doesnt work for now. This is very basic and i will make more interesting tests when i can make it work. 

Can you confirm my helper function `valueAtAddressEquals` in zerohint_utils_test.go is correct?

**Comment by TAdev0** [comment]
Ok it seems that in my implementation of the hint, when i write to memory there is an issue : 

```
			err = vm.Memory.Write(inputsAddress.SegmentIndex, 1, &mvLowLow)
			if err != nil {
				return err
			}
```

Currently , i write to offset 0, 1, 2 and 3 of the segment. If i change these value for random ones (10, 11, 12, 13 for example), it works. Don't know why for now

File: pkg/hintrunner/zero/zerohint_keccak.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+			if err != nil {
+				return err
+			}
+
+			err = vm.Memory.Write(inputsPtr.SegmentIndex, inputsPtr.Offset+2, &mvHighLow)
+			if err != nil {
+				return err
+			}
+
+			err = vm.Memory.Write(inputsPtr.SegmentIndex, inputsPtr.Offset+3, &mvHighHigh)
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_keccak.go:92]: You can just return the ```vm.Memory.Write```

File: pkg/hintrunner/zero/zerohint_keccak.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+			var maxUint64Big big.Int
+			maxUint64Big = *maxUint64Big.SetUint64(math.MaxUint64)
+
+			lowResultBig := new(big.Int).Set(&lowBig)
+			lowResultBigLow := *lowResultBig
+			lowResultBigLow.And(&lowResultBigLow, &maxUint64Big)
+			lowResultFeltLow := new(fp.Element).SetBigInt(&lowResultBigLow)
+			mvLowLow := mem.MemoryValueFromFieldElement(lowResultFeltLow)
+
+			fmt.Println(mvLowLow)
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_keccak.go:51]: rogue print 🤓

File: pkg/hintrunner/zero/zerohint_keccak.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+			lowResultBigHigh := *lowResultBig
+			lowResultBigHigh.Rsh(&lowResultBigHigh, 64)
+			lowResultBigHigh.And(&lowResultBigHigh, &maxUint64Big)
+			lowResultFeltHigh := new(fp.Element).SetBigInt(&lowResultBigHigh)
+			mvLowHigh := mem.MemoryValueFromFieldElement(lowResultFeltHigh)
+
+			fmt.Println(mvLowHigh)
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_keccak.go:59]: rogue print

File: pkg/hintrunner/zero/zerohint_keccak.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+			highResultBig := new(big.Int).Set(&highBig)
+			highResultBigLow := *highResultBig
+			highResultBigLow.And(&highResultBigLow, &maxUint64Big)
+			highResultFeltLow := new(fp.Element).SetBigInt(&highResultBigLow)
+			mvHighLow := mem.MemoryValueFromFieldElement(highResultFeltLow)
+
+			fmt.Println(mvHighLow)
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_keccak.go:67]: rogue print

File: pkg/hintrunner/zero/zerohint_keccak.go [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+			highResulBigHigh := *highResultBig
+			highResulBigHigh.Rsh(&highResulBigHigh, 64)
+			highResulBigHigh.And(&highResulBigHigh, &maxUint64Big)
+			highResultFeltHigh := new(fp.Element).SetBigInt(&highResulBigHigh)
+			mvHighHigh := mem.MemoryValueFromFieldElement(highResultFeltHigh)
+
+			fmt.Println(mvHighHigh)
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_keccak.go:75]: rogue print

File: pkg/hintrunner/zero/zerohint_keccak_test.go [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+					"inputs",
+					[]*fp.Element{
+						feltString("0"),
+						feltString("0"),
+						feltString("1"),
+						feltString("0"),
+					}),
+			},
+			{
+				operanders: []*hintOperander{
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_keccak_test.go:86]: Idk if this test is necessary, as it basically tests the same test as the one with 1 and 1 above. But I guess the more tests the better

File: pkg/hintrunner/zero/zerohint_keccak_test.go [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+					[]*fp.Element{
+						feltUint64(18446744073709551615),
+						feltUint64(18446744073709551615),
+						feltUint64(18446744073709551615),
+						feltUint64(18446744073709551615),
+					}),
+			},
+		},
+	})
+}
```
Comment (MaksymMalicki) [inline pkg/hintrunner/zero/zerohint_keccak_test.go:195]: You could add one more test, where all the result values are different than 0 and 1

**Comment by MaksymMalicki** [comment]
The PR generally looks very good, there are only some nitpicks, additional tests and rogue prints. In case of this PR, as all the values are unsigned, instead of using the ```big.int``` library you can try and use the ```uint256``` library, which is 3-10x faster. All the logic is already implemented, so changing that should be fairly simple. Here is a simple example of how to load felt <-> uint256 and perform some logic operations.
```
a = new(fp.Element).SetUint64(100)
aUint256 := uint256.Int(a.Bits())
bUint256 := new(uint256.Int).Lsh(uint256.NewInt(1), 127)
cUint256 := new(uint256.Int).And(&aUint256, bUint256).Bytes32()
bFelt, err := fp.BigEndian.Element(&cUint256)
if err != nil {
return err
}
fmt.Println(bFelt)
```

File: pkg/hintrunner/zero/zerohint_keccak.go (hunk 3 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_keccak.go:51]: oups
thanks!

File: pkg/hintrunner/zero/zerohint_keccak.go (hunk 2 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_keccak.go:92]: yes indeed, thanks!

File: pkg/hintrunner/zero/zerohint_keccak_test.go (hunk 7 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_keccak_test.go:86]: true, will remove it and add more pertinent tests for edge cases

File: pkg/hintrunner/zero/zerohint_keccak_test.go (hunk 8 above)
Comment (TAdev0) [reply pkg/hintrunner/zero/zerohint_keccak_test.go:195]: done!

**Comment by TAdev0** [comment]
@MaksymMalicki  comments addressed. I replaced usage of Big Int with uint256.  I added tests and removed the redondant one. Should be good!

**Comment by TAdev0** [comment]
Ready to be merged @cicr99  @rodrigo-pino

**Review by rodrigo-pino — APPROVED** [review]
